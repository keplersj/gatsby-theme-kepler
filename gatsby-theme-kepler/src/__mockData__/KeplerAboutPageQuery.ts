import { AboutPageQuery } from "../pages/about";

export const KeplerAboutPageQuery: AboutPageQuery = {
  biography: {
    childMdx: {
      frontmatter: {
        title: "Biography"
      },
      body: ""
    }
  },
  experience: {
    edges: [
      {
        node: {
          id: "example-experience-listing",
          childMdx: {
            id: "example-experience-listing-mdx",
            body: "",
            frontmatter: {
              title: "Example Workplace",
              position: "Sample Position",
              /* eslint-disable @typescript-eslint/camelcase */
              rawEnd: "2018-12-18",
              end_date: "Dec 2018",
              rawStart: "2017-12-18",
              start_date: "Dec 2017"
              /* eslint-enable @typescript-eslint/camelcase */
            }
          }
        }
      }
    ]
  },
  education: {
    edges: [
      {
        node: {
          id: "example-education-listing",
          childMdx: {
            id: "example-education-listing-mdx",
            body: "",
            frontmatter: {
              title: "Example Institution",
              degree: "Sample Degree",
              /* eslint-disable @typescript-eslint/camelcase */
              rawEnd: "2018-12-18",
              end_date: "Dec 2018",
              rawStart: "2017-12-18",
              start_date: "Dec 2017"
              /* eslint-enable @typescript-eslint/camelcase */
            }
          }
        }
      }
    ]
  },
  skills: {
    childMdx: {
      frontmatter: {
        title: "Skills"
      },
      body: ""
    }
  }
};
