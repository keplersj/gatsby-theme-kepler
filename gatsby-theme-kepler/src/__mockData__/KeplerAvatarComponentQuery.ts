import { AvatarComponentQuery } from "../components/Avatar";

export const KeplerAvatarComponentQuery: AvatarComponentQuery = {
  image: {
    childImageSharp: {
      sqip: { dataURI: "data+image/svg+xml" },
      fixed: {
        width: 200,
        height: 200,
        src: "/static/example/1/avatar.jpg",
        srcSet: `/static/example/2/avatar.jpg 1x,
                /static/example/3/avatar.jpg 1.5x,
                /static/example/4/avatar.jpg 2x`,
      },
    },
  },
};
