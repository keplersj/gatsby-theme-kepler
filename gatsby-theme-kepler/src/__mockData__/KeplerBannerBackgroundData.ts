import { BannerData } from "../components/BannerBackground";

const sampleBase64 =
  "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAPABQDASIAAhEBAxEB/8QAGAAAAgMAAAAAAAAAAAAAAAAAAAYBAgX/xAAVAQEBAAAAAAAAAAAAAAAAAAAAAf/aAAwDAQACEAMQAAAB1ZUrU0CyL//EABoQAAICAwAAAAAAAAAAAAAAAAACARIREyH/2gAIAQEAAQUCiFMIUg6XY2Mf/8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAwEBPwE//8QAFhEBAQEAAAAAAAAAAAAAAAAAAAER/9oACAECAQE/AW1//8QAGBAAAwEBAAAAAAAAAAAAAAAAAAIyEDH/2gAIAQEABj8C3pbFsf/EAB4QAAEDBAMAAAAAAAAAAAAAAAABEUEhUWFxkdHx/9oACAEBAAE/IcnggdtoLfKJjfce4f/aAAwDAQACAAMAAAAQGw//xAAWEQEBAQAAAAAAAAAAAAAAAAABABH/2gAIAQMBAT8QAsL/xAAVEQEBAAAAAAAAAAAAAAAAAAABEP/aAAgBAgEBPxBWH//EABwQAQACAgMBAAAAAAAAAAAAAAEAESFRcbHRwf/aAAgBAQABPxBpi3Fuos5FpCIbpBzBcLAlYNX9gZ9/s//Z";

export const KeplerBannerBackgroundData: BannerData = {
  banner: {
    childImageSharp: {
      sqip: { dataURI: "data+image/svg+xml" },
      highQuality: {
        base64: sampleBase64,
        src: "/static/banner_example/3/banner.jpg",
        srcWebp: "/static/banner_example/4/banner.webp"
      },
      lowQuality: {
        base64: sampleBase64,
        src: "/static/banner_example/1/banner.jpg",
        srcWebp: "/static/banner_example/2/banner.jpg"
      }
    }
  },
  bannerDark: {
    childImageSharp: {
      sqip: { dataURI: "data+image/svg+xml" },
      highQuality: {
        base64: sampleBase64,
        src: "/static/banner_dark_example/3/banner_dark.jpg",
        srcWebp: "/static/banner_dark_example/4/banner_dark.webp"
      },
      lowQuality: {
        base64: sampleBase64,
        src: "/static/banner_example/1/banner_dark.jpg",
        srcWebp: "/static/banner_example/2/banner_dark.webp"
      }
    }
  }
};
