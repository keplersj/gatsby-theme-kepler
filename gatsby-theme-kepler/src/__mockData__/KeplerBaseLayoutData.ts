import { BaseLayoutData } from "../components/BaseLayout";

export const KeplerBaseLayoutData: BaseLayoutData = {
  file: {
    childImageSharp: {
      fixed: {
        base64: "",
        width: 100,
        height: 100,
        src: "",
        srcSet: ""
      }
    }
  },
  site: {
    siteMetadata: {
      title: "Kepler Sticka-Jones",
      description: "",
      siteUrl: "https://site.dev",
      social: [
        {
          name: "Twitter",
          id: "@test",
          url: "https://twitter.com/test"
        },
        {
          name: "GitHub",
          id: "test",
          url: "https://github.com/test"
        }
      ]
    }
  }
};
