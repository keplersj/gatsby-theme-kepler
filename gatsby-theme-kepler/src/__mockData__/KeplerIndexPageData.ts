import { IndexPageData } from "../pages/index";

export const KeplerIndexPageData: IndexPageData = {
  site: {
    siteMetadata: {
      title: "Example Site",
      description: "This is an example of a personal web site.",
      siteUrl: "https://site.dev",
      social: [
        {
          name: "Email",
          id: "test@example.dev",
          url: "mailto:test@example.dev",
          isProfile: false
        },
        {
          name: "PGP",
          id: "XXXX XXXX XXXX XXXX XXXX XXXX XXXX XXXX XXXX XXXX XXXX",
          url: "https://example.dev/pub.asc",
          isProfile: false
        },
        {
          name: "GitHub",
          id: "test",
          url: "https://github.dev/test",
          isProfile: true
        },
        {
          name: "Twitter",
          id: "@test",
          url: "https://twitter.dev/test",
          isProfile: true
        },
        {
          name: "LinkedIn",
          id: "test",
          url: "https://linkedin.dev/in/test",
          isProfile: true
        },
        {
          name: "Keybase",
          id: "test",
          url: "https://keybase.dev/test",
          isProfile: true
        },
        {
          name: "Instagram",
          id: "@test",
          url: "https://instagram.dev/test",
          isProfile: true
        },
        {
          name: "Unknown Platform of the Future",
          id: "test",
          url: "https://ü§∑üèº‚Äç‚ôÇÔ∏è.dev/test",
          isProfile: true
        }
      ],
      nav: [
        {
          name: "Blog",
          url: "/blog"
        },
        {
          name: "Portfolio",
          url: "/portfolio"
        },
        {
          name: "About",
          url: "/about"
        }
      ]
    }
  },
  biography: {
    childMdx: {
      frontmatter: {
        title: "Biography"
      },
      body:
        "I am a computer programmer and college student based out of Salt Lake City, with experience in entrepreneurship, student leadership, and open source software development."
    }
  },
  metadataImage: {
    childImageSharp: {
      fixed: {
        base64: "",
        width: 100,
        height: 100,
        src: "",
        srcSet: ""
      }
    }
  },
  blogPosts: {
    edges: [
      {
        node: {
          id: "post-1",
          title: "Test Post (without Image)",
          slug: "/blog/01-post-without-image",
          date: "Dec 18, 2018",
          rawDate: "2018-12-18",
          excerpt: "This is an example of a blog post without a featured image."
        }
      },
      {
        node: {
          id: "post-2",
          title: "Test Post (with Image)",
          slug: "/blog/02-post-with-image",
          date: "Dec 18, 2019",
          rawDate: "2018-12-18",
          excerpt: "This is an example of a blog post with a featured image.",
          featuredImage: {
            childImageSharp: {
              fluid: {
                aspectRatio: 1.0,
                src: "/static/image-1/image.jpeg",
                srcSet: "/static/image-1/image.jpeg 1x",
                sizes: "(min-width 100px)"
              }
            }
          }
        }
      }
    ]
  }
};
