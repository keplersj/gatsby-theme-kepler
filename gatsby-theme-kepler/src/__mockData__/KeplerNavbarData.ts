import { NavbarData } from "../components/Navbar";

export const KeplerNavbarData: NavbarData = {
  site: {
    siteMetadata: {
      title: "Kepler Sticka-Jones",
      nav: [
        {
          name: "Blog",
          url: "/blog"
        },
        {
          name: "Projects",
          url: "/projects"
        },
        {
          name: "About",
          url: "/about"
        }
      ]
    }
  }
};
