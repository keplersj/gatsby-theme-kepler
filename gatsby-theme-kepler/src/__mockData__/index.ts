export { KeplerAvatarComponentQuery } from "./KeplerAvatarComponentQuery";
export { KeplerBannerBackgroundData } from "./KeplerBannerBackgroundData";
export { KeplerBaseLayoutData } from "./KeplerBaseLayoutData";
export { KeplerNavbarData } from "./KeplerNavbarData";
