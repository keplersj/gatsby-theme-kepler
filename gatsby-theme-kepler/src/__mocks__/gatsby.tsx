import * as React from "react";
const gatsby = jest.requireActual("gatsby");

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn(
    ({
      // these props are invalid for an `a` tag
      /* eslint-disable @typescript-eslint/no-unused-vars */
      /* eslint-disable no-unused-vars */
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      /* eslint-enable @typescript-eslint/no-unused-vars */
      /* eslint-enable no-unused-vars */
      to,
      ...rest
      // eslint-disable-next-line jsx-a11y/anchor-has-content
    }): React.ReactElement => <a {...rest} href={to} />
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn()
};
