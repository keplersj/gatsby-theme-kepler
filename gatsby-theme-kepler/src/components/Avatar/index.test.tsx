import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import { KeplerAvatarComponentQuery } from "../../__mockData__/KeplerAvatarComponentQuery";
import { Avatar } from ".";

beforeEach(() => {
  (useStaticQuery as jest.Mock).mockImplementation(
    (): object => KeplerAvatarComponentQuery
  );
});

describe("Avatar Component", () => {
  it("renders as expected", () => {
    const tree = renderer.create(<Avatar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
