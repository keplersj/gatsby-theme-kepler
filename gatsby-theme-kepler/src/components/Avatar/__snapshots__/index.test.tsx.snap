// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component renders as expected 1`] = `
.emotion-0 {
  margin-bottom: 10px;
  border-radius: 140px;
}

<div
  aria-label="Avatar"
  role="img"
>
  <div
    className="emotion-0 gatsby-image-wrapper"
    style={
      Object {
        "display": "inline-block",
        "height": 200,
        "overflow": "hidden",
        "position": "relative",
        "width": 200,
      }
    }
  >
    <img
      alt=""
      src="data+image/svg+xml"
      style={
        Object {
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "center",
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "transitionDelay": "500ms",
          "width": "100%",
        }
      }
    />
    <picture>
      <source
        srcSet="/static/example/2/avatar.jpg 1x,
                /static/example/3/avatar.jpg 1.5x,
                /static/example/4/avatar.jpg 2x"
      />
      <img
        alt="Avatar"
        height={200}
        loading="lazy"
        onLoad={[Function]}
        src="/static/example/1/avatar.jpg"
        srcSet="/static/example/2/avatar.jpg 1x,
                /static/example/3/avatar.jpg 1.5x,
                /static/example/4/avatar.jpg 2x"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 500ms",
            "width": "100%",
          }
        }
        width={200}
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "<picture><source srcset=\\"/static/example/2/avatar.jpg 1x,
                /static/example/3/avatar.jpg 1.5x,
                /static/example/4/avatar.jpg 2x\\" /><img loading=\\"lazy\\" width=\\"200\\" height=\\"200\\" srcset=\\"/static/example/2/avatar.jpg 1x,
                /static/example/3/avatar.jpg 1.5x,
                /static/example/4/avatar.jpg 2x\\" src=\\"/static/example/1/avatar.jpg\\" alt=\\"Avatar\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
        }
      }
    />
  </div>
</div>
`;
